### PR TITLE
Fix redis-namespace deprecations

### PR DESCRIPTION
**Pull request by carsonreinke**
Fixes part of https://github.com/resque/resque-scheduler/issues/600

### CODE REVIEW

File: lib/resque/scheduler/lock/resilient.rb [hunk 1]
```diff
@@ -43,7 +43,7 @@ def locked_sha(refresh = false)
           @locked_sha = nil if refresh
 
           @locked_sha ||=
-            Resque.redis.script(:load, <<-EOF.gsub(/^ {14}/, ''))
+            Resque.redis.redis.script(:load, <<-EOF.gsub(/^ {14}/, ''))
```
Comment (jaredbeck) [inline lib/resque/scheduler/lock/resilient.rb]: hmmm needs more redis' ;)

Comment (carsonreinke) [inline lib/resque/scheduler/lock/resilient.rb]: I know.

```
2.3.3 :001 > Resque.redis.class
 => Resque::DataStore 
2.3.3 :002 > Resque.redis.redis.class
 => Redis 
```

Calling `Resque.redis.script` delegates to `Redis::Namespace`, `Resque.redis.redis.script` delegates to `Redis`.  Scripts do not need a namespace.

**Comment by josephpage** [comment]
Method `Resque::DataStore#redis` relies on [`Resque::DataStore#method_missing`](http://www.rubydoc.info/gems/resque/Resque/DataStore#method_missing-instance_method)


So `Resque::DataStore#redis` returns the return of `Redis::Namespace#redis`. Maybe we should use `Redis::Namespace#_client` instead, as suggested in the code : https://github.com/resque/redis-namespace/blob/152253f5d38fcf325f288e881e7ade57c0e346bd/lib/redis/namespace.rb#L255

BTW, I am glad that this PR is merged in the near future.

**Comment by carsonreinke** [comment]
Fixes #632 and @josephpage suggestion for another deprecation.

**Pull request merged**